src/webpimage.cpp

```diff
@@ -489,15 +489,16 @@ namespace Exiv2 {
         DataBuf chunkId(5);
         chunkId.pData_[4] = '\0' ;
 
-        io_->read(data, WEBP_TAG_SIZE * 3);
+        io_->readOrThrow(data, WEBP_TAG_SIZE * 3, Exiv2::kerCorruptedMetadata);
 
-        const uint32_t filesize = Exiv2::getULong(data + WEBP_TAG_SIZE, littleEndian) + 8;
-        enforce(filesize <= io_->size(), Exiv2::kerCorruptedMetadata);
+        const long filesize = Exiv2::getULong(data + WEBP_TAG_SIZE, littleEndian) + 8;
```

Comment:
Add a `#include "safe_op.hpp"` at the top of the file and remove the check in the following line. `Safe:add` catches overflow errors for you:
```suggestion
        const long filesize = Safe::add(Exiv2::getULong(data + WEBP_TAG_SIZE, littleEndian), 8);
```